snowplow-tracker/src/main/java/com/snowplowanalytics/snowplow/internal/tracker/Tracker.java

```diff
@@ -540,6 +552,14 @@ private Tracker(@NonNull TrackerBuilder builder) {
         this.level = builder.logLevel;
         this.foregroundTimeout = builder.foregroundTimeout;
         this.backgroundTimeout = builder.backgroundTimeout;
+        this.trackerVersionSuffix = builder.trackerVersionSuffix;
+
+        if (trackerVersionSuffix != null) {
+            String suffix = trackerVersionSuffix.replaceAll("[^A-Za-z0-9.\\-]", "");
```

Comment:
Any reason we're allowing `\` on Android but not on iOS?

Reply:
On iOS I used a different approach without the regex. Here `\` is the escape for the other `\` that is the escape for the dash `-`, which is a special character in that regex. Bit crazy!

Reply:
You don't need to escape the `\` for the `-`. By adding two `\` you are escaping the `\`. Infact, you don't need to escape `-` at all if you place it at the end of the character class. You can escape it (since it does have a special meaning in a character class) but I actually think it's more conventional to not escape it and place it at the end.
The regex you want is: `[^A-Za-z0-9.-]`
https://regexr.com/620jo

Reply:
Azz ... I fix it and release it